### PR TITLE
feat: conversation rename event handling (AR-2483)

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We are adding support for conversation rename event and with that support incremental sync to speed up the app.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt [hunk 1]
```diff
@@ -384,6 +384,26 @@ internal class ConversationEventReceiverImpl(
         }
     }
 
+    private suspend fun handleRenamedConversation(event: Event.Conversation.RenamedConversation) {
+        conversationRepository.updateConversationName(event.conversationId, event.conversationName, event.timestampIso)
+            .onSuccess {
+                kaliumLogger.withFeatureId(EVENT_RECEIVER).d("$TAG - The Conversation was renamed: ${event.conversationName}")
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt:390]: Log policy is here licence and registration please 🚔
maybe it's not a great idea to log the conversation name

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/Messages.sq [hunk 2]
```diff
@@ -128,6 +128,16 @@ CREATE TABLE MessageMissedCallContent (
       PRIMARY KEY (message_id, conversation_id)
 );
 
+CREATE TABLE MessageConversationChangedContent (
+      message_id TEXT NOT NULL,
+      conversation_id TEXT AS QualifiedIDEntity NOT NULL,
+
```
Comment (MohamadJaara) [inline persistence/src/commonMain/db_user/com/wire/kalium/persistence/Messages.sq:134]: i guess we need to store the id of the user who changed the name

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt (hunk 1 above)
Comment (tmspzz) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt:390]: @yamilmedina please censor the name of the conversation. You can ask @MohammedMokresh how to

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/Messages.sq (hunk 2 above)
Comment (yamilmedina) [reply persistence/src/commonMain/db_user/com/wire/kalium/persistence/Messages.sq:134]: This is stored in the `Messages` parent table, so this is only related for this log kind of system messages :D

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt (hunk 1 above)
Comment (yamilmedina) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt]: done !

**Review by Garzas — APPROVED** [review]
That was fast 😲 
LGTM 🚀

**Pull request merged**